### PR TITLE
Introduces 'cpt_mode' to gff3_rebase tool

### PR DESCRIPTION
**Pull request by curtisim0**
Currently, the rebase tool keys on the gene; and the "cpt_mode" will introduce the ability to key on the CDS.

This tool originated in the CPT and has drifted in some regards, and this allows the drift to still be respected as well as the CPT-galaxy-using folks to still be able to observe expected rebasing patterns.

Verbage and clarity will be improved for the tool section later by @jgill.

FOR CONTRIBUTOR:
* [x] - I have read the [CONTRIBUTING.md](https://github.com/galaxyproject/tools-iuc/blob/master/CONTRIBUTING.md) document and this tool is appropriate for the tools-iuc repo.
* [x] - License permits unrestricted use (educational + commercial)
* [ ] - This PR adds a new tool or tool collection
* [x] - This PR updates an existing tool or tool collection
* [ ] - This PR does something else (explain below)

### CODE REVIEW

File: tools/gff3_rebase/gff3_rebase.xml [hunk 1]
```diff
@@ -6,7 +6,11 @@
   <expand macro="requirements"/>
   <version_command>python gff3_rebase.py --version</version_command>
   <command detect_errors="aggressive"><![CDATA[
-    python '$__tool_directory__/gff3_rebase.py'
+    #if $cpt_mode
+      python '$__tool_directory__/cpt_gff3_rebase.py'
+    #else
+      python '$__tool_directory__/gff3_rebase.py'
+    #end if
```
Comment (curtisim0) [inline tools/gff3_rebase/gff3_rebase.xml:13]: ehhhhhh, I dunno

This is okay.

The issue I was running into locally was getting BCBio to play nice with the "CPT gff" parser. Which, I decided to pull out; and try just to use BCbio. I was struggling to get (I think) planemo to flush the dependencies, and the error that was happening _kept happening_.

I'd like to see if CI still fusses in the same way...

File: tools/gff3_rebase/gff3_rebase.xml [hunk 2]
```diff
@@ -76,7 +87,12 @@ which will allow you to display the results in the correct location in visualiza
 
 **Options**
 
-There are two optional flags which can be passed.
+The cpt_mode flag rebases to the CDS instead of the Gene. 
+This is common for phage genomes, so that RBS & Shine-Dalgarno sequences can be
+appropriately respected.
+However, this is set to OFF by default.
+
+There are two additional optional flags which can be passed.
```
Comment (curtisim0) [inline tools/gff3_rebase/gff3_rebase.xml:95]: Verbage needs love from jgill

File: tools/gff3_rebase/gff3_rebase.xml [hunk 3]
```diff
@@ -15,22 +19,29 @@
...
-          <param name="parent" value="parent.gff"/>
-          <param name="child" value="child.gff"/>
-          <param name="protein2dna" value="True" />
-          <output name="output" file="proteins.gff"/>
-      </test>
+    <test>
+      <param name="cpt_mode" value="True"/>
+      <param name="parent" value="parent.gff"/>
+      <param name="child" value="child.gff"/>
+      <output name="log_output" file="cpt_log_output.txt"/>
```
Comment (curtisim0) [inline tools/gff3_rebase/gff3_rebase.xml:37]: basically was trying to "capture" that the cpt_mode was called and also that it doesn't break the other test 🤷 

I need muh assert_called_once...

**Review by curtisim0 — COMMENTED** [review]
@hexylena I dunno...this might work just fine...it just has a bit of a code smell to me 😅  

But, I tried to strip out "cpt-isms" of this (Cpt parser) and I'm not quite sure if it's going to work the same.

Another option is to honestly get a "cpt_gff3_rebaser" up on the phage eu galaxy.

Regardless, happy to try and get this to work 👍

**Comment by hexylena** [comment]
@curtisim0 I've looked at the changes from the CPT Gff3 parser, and since you switched it to use bcbio-gff, the diff is extremely minimal, let's just merge them completely. I'll push the changes directly.

I've done the following

- stolen CPT's test cases to be sure it works like y'all expect
- updated test cases (small differences between cpt gff parser, bcbio, namely the lack of a sequence region/presence of trailing `;` which I don't think is really correct gff3 but it's also relatively common)
- merged the scripts
- updated our existing test data (it still comes out the same.)

I'm not seeing where it rebases to the cds instead of the gene, maybe that got lost (edit: no, not seeing that in the upstream source either), could you point me to where that's happening?

**Comment by curtisim0** [comment]
> could you point me to where that's happening?
It's a good question. I could not find _where_ that was happening, hence why I tried to do it this way to determine if it honestly was any different. I might doubly-confirm with Jason that _this is the culprit_, since it's also not clear to you.

Standby for comment 🤠

**Review by bgruening — APPROVED** [review]

**Comment by bgruening** [comment]
ping @hexylena

**Comment by hexylena** [comment]
@bgruening hmm?

it's not ready yet, see @curtisim0's last comment: "Standby for comment 🤠"

I'm the assignee on this, no worries, will merge it when it's ready :)

**Comment by bgruening** [comment]
ping @curtisim0

**Comment by bgruening** [comment]
ping @curtisim0 @hexylena

**Comment by hexylena** [comment]
I guess let's close it for now. @curtisim0 if there's a change in this let me know!

**Comment by curtisim0** [comment]
I will keep y'all posted _when I know_ what we're doing.

Haven't had the ability to balance this change with the other stuff going on.

Thanks!

**Comment by hexylena** [comment]
no worries mate, there isn't any rush

**Pull request closed without merging**